static/js/apps/eval_rig/eval_list.tsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Component to show the full list of evaluation queries */
+
+import { User } from "firebase/auth";
+import React, { useState } from "react";
+import { Button, Input, Modal } from "reactstrap";
+
+import { getCallExistence } from "./data_store";
+import { DcCall } from "./eval_section";
+import { Query } from "./query_section";
+
+interface EvalListPropType {
+  user: User;
+  sheetId: string;
+  queries: Record<string, Query>;
+  calls: Record<string, DcCall[]>;
+  onQuerySelected: (query: Query) => void;
+}
+
+export function EvalList(props: EvalListPropType): JSX.Element {
+  const [userEvalsOnly, setUserEvalsOnly] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [queryCompletionStatus, setQueryCompletionStatus] = useState({});
+
+  const userLdap = props.user.email.split("@")[0];
+  const orderedQueries = Object.keys(props.queries)
+    .sort()
+    .map((queryId) => props.queries[queryId]);
+
+  const openModal = () => {
+    setModalOpen(true);
+    const existPromises = orderedQueries.map((query) => {
+      return Promise.all(
+        props.calls[query.id].map((call) => {
+          return getCallExistence(props.sheetId, query.id, call.id);
+        })
+      );
+    });
```

Comment:
is there a way to count the collections under query.id in firestore? then can just do one query and check if the count is the same as the total count?

Reply:
unfortunately not, I've tried that and I just get 0 ..

Reply:
I wonder if this would work https://firebase.google.com/docs/firestore/query-data/aggregation-queries#use_the_count_aggregation